src/controller.js

```diff
@@ -351,6 +352,43 @@ class Controller {
 
     return request;
   }
+
+  /**
+   * Generate an optionally signed GET URL for an object
+   */
+  async generateGetUrl(opts) {
```

Comment:
Honestly, I don't like reimplementing features from `aws-sdk`, someone has to maintain the code too..
There is probably no bugs here, but when we did `fast-azure-storage` we found a lot of one-off bugs, related to character encoding and other stupid stuff... If we can avoid we really should.